group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/RangeAssignor.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.assignor;
+
+import org.apache.kafka.common.Uuid;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static java.lang.Math.min;
+
+/**
+ * The Server Side Sticky Range Assignor inherits properties of both the range assignor and the sticky assignor.
+ * Properties are as follows:
+ * <ol>
+ * <li> Each member must get at least one partition for every topic that it is subscribed to. The only exception is when
+ * the number of subscribed members is greater than the number of partitions for that topic. (Range) </li>
+ * <li> Partitions should be assigned to members in a way that facilitates the join operation when required. (Range) </li>
+ * This can only be done if every member is subscribed to the same topics and the topics are co-partitioned.
+ * Two streams are co-partitioned if the following conditions are met:
+ * <ul>
+ * <li> The keys must have the same schemas.
+ * <li> The topics involved must have the same number of partitions.
+ * </ul>
+ * <li> Members should retain as much of their previous assignment as possible to reduce the number of partition movements during reassignment. (Sticky) </li>
+ * </ol>
+ */
+public class RangeAssignor implements PartitionAssignor {
+
+    private static final Logger log = LoggerFactory.getLogger(RangeAssignor.class);
+
+    public static final String RANGE_ASSIGNOR_NAME = "range";
+
+    @Override
+    public String name() {
+        return RANGE_ASSIGNOR_NAME;
+    }
+
+    static class RemainingAssignmentsForMember<String, Integer> {
```

Comment:
nit: `<String, Integer>` does not seem to be required.

Reply:
removed, since it was derived from a generic pair class, missed removing it, thanks for the catch!